### PR TITLE
updated frontend to display nft gallery from 3box/opensea

### PR DESCRIPTION
**Pull request by dan13ram**
we must first merge my previous PR #221 that sets up the backend for this PR before merging this PR.

### CODE REVIEW

**Review by pakokrew — APPROVED** [review]
LGTM

File: packages/web/components/Player/Box/PlayerGallery.tsx [hunk 1]
```diff
@@ -1,63 +1,149 @@
...
+}) => (
+  <Link
+    href={nft.permaLink}
+    isExternal
+    mb={noMargin ? undefined : 6}
+    minW={0}
+    display="flex"
+  >
+    <HStack spacing={6}>
+      <Flex width="126px" height="126px">
```
Comment (hammadj) [inline packages/web/components/Player/Box/PlayerGallery.tsx:37]: Sizing is hard coded as a magic number. Can we use responsive styles or one of the standard sizes from the theme instead for consistency: https://chakra-ui.com/docs/theming/theme#sizes ?

File: packages/web/components/Player/Box/PlayerGallery.tsx [hunk 2]
```diff
@@ -1,63 +1,149 @@
...
+        <Box
+          bgImage={`url(${nft.imageUrl})`}
```
Comment (hammadj) [inline packages/web/components/Player/Box/PlayerGallery.tsx:39]: We should start using the new NextJS Image component: https://nextjs.org/docs/basic-features/image-optimization

No need to implement in this PR, can do it in a PR specifically to setup the Image component

File: packages/web/components/Player/Box/PlayerGallery.tsx [hunk 3]
```diff
@@ -1,63 +1,149 @@
...
+          m="auto"
+        />
+      </Flex>
+      <Flex direction="column">
+        <Text
+          fontSize="xs"
+          fontFamily="heading"
+          mt={3}
+          mb={3}
+          casing="uppercase"
```
Comment (hammadj) [inline packages/web/components/Player/Box/PlayerGallery.tsx:54]: Why not use the Heading component instead?

File: packages/web/components/Player/Box/PlayerGallery.tsx [hunk 4]
```diff
@@ -1,63 +1,149 @@
...
-              {nft.priceInEth}Ξ ($
-              {parseFloat(`${nft.priceInEth * fakeEthPrice}`).toFixed(2)})
-            </Text>
-          </Box>
-        </HStack>
-      ))}
-      {(fakeData || []).length > 3 && (
+      {!loading &&
+        favorites &&
+        favorites.map((nft) => <GalleryItem nft={nft} key={nft.tokenId} />)}
```
Comment (hammadj) [inline packages/web/components/Player/Box/PlayerGallery.tsx:77]:
```suggestion
        favorites?.map((nft) => <GalleryItem nft={nft} key={nft.tokenId} />)}
```

File: packages/web/components/Player/Box/PlayerGallery.tsx [hunk 5]
```diff
@@ -1,63 +1,149 @@
...
+      {!loading && data && data.length > 3 && (
```
Comment (hammadj) [inline packages/web/components/Player/Box/PlayerGallery.tsx:78]:
```suggestion
      {!loading && data?.length > 3 && (
```

File: packages/web/components/Player/Box/PlayerGallery.tsx [hunk 6]
```diff
@@ -1,63 +1,149 @@
...
+      <Modal
+        isOpen={isOpen}
+        onClose={onClose}
+        isCentered
+        scrollBehavior="inside"
+      >
+        <ModalOverlay css={{ backdropFilter: 'blur(8px)' }}>
+          <ModalContent maxW="75rem" bg="none">
+            <Box
+              bg="rgba(70, 20, 100, 0.8)"
```
Comment (hammadj) [inline packages/web/components/Player/Box/PlayerGallery.tsx:99]: Should use values from the theme

File: packages/web/components/Player/Box/PlayerGallery.tsx [hunk 7]
```diff
@@ -1,63 +1,149 @@
...
         </Text>
       )}
+      <Modal
+        isOpen={isOpen}
+        onClose={onClose}
+        isCentered
+        scrollBehavior="inside"
+      >
+        <ModalOverlay css={{ backdropFilter: 'blur(8px)' }}>
+          <ModalContent maxW="75rem" bg="none">
```
Comment (hammadj) [inline packages/web/components/Player/Box/PlayerGallery.tsx:97]: should use theme values
```suggestion
          <ModalContent maxW="6xl" bg="none">
```

File: packages/web/components/Player/Box/PlayerGallery.tsx [hunk 8]
```diff
@@ -1,63 +1,149 @@
...
+                  fontSize="sm"
+                  fontWeight="bold"
+                  color="blueLight"
+                  as="div"
+                  mr="auto"
+                >
+                  Gallery
+                </Text>
+                <FaTimes
+                  color="#A5B9F6"
```
Comment (hammadj) [inline packages/web/components/Player/Box/PlayerGallery.tsx:116]: Should use values from the theme and also use the Icon component from Chakra: https://chakra-ui.com/docs/components/icon#using-a-third-party-icon-library

File: packages/web/components/Player/Box/PlayerGallery.tsx (hunk 3 above)
Comment (dan13ram) [reply packages/web/components/Player/Box/PlayerGallery.tsx:54]: Will do. I just edited the previous code.

**Review by hammadj — CHANGES_REQUESTED** [review]
Good stuff! just a few minor nits

File: packages/web/components/Player/Box/PlayerGallery.tsx (hunk 8 above)
Comment (dan13ram) [reply packages/web/components/Player/Box/PlayerGallery.tsx:116]: Will replace react-icons with chakra-icons in a separate PR. Here I'm just reusing the same icon from other existing components.

File: packages/web/components/Player/Box/PlayerGallery.tsx (hunk 2 above)
Comment (dan13ram) [reply packages/web/components/Player/Box/PlayerGallery.tsx:39]: Sounds good.

**Comment by hammadj** [comment]
@dan13ram Something is wrong with the prices, it's showing many of them with the same price: 
image

**Comment by dan13ram** [comment]
@hammadj Good catch! There's a mistake in the logic. Fixing it!

**Comment by dan13ram** [comment]
@hammadj have made the changes you requested. 
I believe its much cleaner now with the help of types from `opensea-js`, and we are showing the price for the lastSale of the NFT.
will make the icons change in another PR, because `react-icons` is being used in multiple places and I think we should remove that unwanted dependency completely.

**Pull request merged**